components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java

```diff
@@ -1320,6 +1320,47 @@ public Policy[] getPolicies(int tenantId, String level) throws APIManagementExce
         return policies;
     }
 
+    /**
+     * Get Policy with corresponding name and type.
+     *
+     * @param tenantId tenantId
+     * @param level    policy type
+     * @param name     policy name
+     * @return Policy with corresponding name and type
+     * @throws APIManagementException
+     */
+    @Override public Policy getPolicyByNameAndType(int tenantId, String level, String name)
+            throws APIManagementException {
+
+        Policy policy = null;
+
+        if (PolicyConstants.POLICY_LEVEL_API.equals(level)) {
+            policy = apiMgtDAO.getAPIPolicy(name, tenantId);
+        } else if (PolicyConstants.POLICY_LEVEL_APP.equals(level)) {
+            policy = apiMgtDAO.getApplicationPolicy(name, tenantId);
+        } else if (PolicyConstants.POLICY_LEVEL_SUB.equals(level)) {
+            policy = apiMgtDAO.getSubscriptionPolicy(name, tenantId);
+        } else if (PolicyConstants.POLICY_LEVEL_GLOBAL.equals(level)) {
+            policy = apiMgtDAO.getGlobalPolicy(name);
+        }
+
+        //Get the API Manager configurations and check whether the unlimited tier is disabled. If disabled, remove
+        // the tier from the array.
+        APIManagerConfiguration apiManagerConfiguration = ServiceReferenceHolder.getInstance()
+                .getAPIManagerConfigurationService().getAPIManagerConfiguration();
+        ThrottleProperties throttleProperties = apiManagerConfiguration.getThrottleProperties();
+
+        if (APIConstants.UNLIMITED_TIER.equals(policy.getPolicyName())) {
+            if (throttleProperties.isEnableUnlimitedTier()) {
+                return policy;
```

Comment:
Rather than having 3 returns, shall we return the null if !throttleProperties.isEnableUnlimitedTier() and if not return policy? Can there be any other instance that returns a null?